src/confio/proofs.ts

```diff
@@ -308,81 +345,95 @@ export interface BatchProof {
 
 /** Use BatchEntry not CommitmentProof, to avoid recursion */
 export interface BatchEntry {
-  exist?: ExistenceProof | undefined;
-  nonexist?: NonExistenceProof | undefined;
+  exist?: ExistenceProof;
+  nonexist?: NonExistenceProof;
 }
-
 export interface CompressedBatchProof {
   entries: CompressedBatchEntry[];
   lookupInners: InnerOp[];
 }
 
 /** Use BatchEntry not CommitmentProof, to avoid recursion */
 export interface CompressedBatchEntry {
-  exist?: CompressedExistenceProof | undefined;
-  nonexist?: CompressedNonExistenceProof | undefined;
+  exist?: CompressedExistenceProof;
+  nonexist?: CompressedNonExistenceProof;
 }
-
 export interface CompressedExistenceProof {
   key: Uint8Array;
   value: Uint8Array;
-  leaf?: LeafOp;
+  leaf: LeafOp;
+
   /** these are indexes into the lookup_inners table in CompressedBatchProof */
   path: number[];
 }
-
 export interface CompressedNonExistenceProof {
   /** TODO: remove this as unnecessary??? we prove a range */
   key: Uint8Array;
-  left?: CompressedExistenceProof;
-  right?: CompressedExistenceProof;
+  left: CompressedExistenceProof;
+  right: CompressedExistenceProof;
 }
 
 function createBaseExistenceProof(): ExistenceProof {
-  return { key: new Uint8Array(), value: new Uint8Array(), leaf: undefined, path: [] };
+  return {
+    key: new Uint8Array(),
+    value: new Uint8Array(),
+    leaf: undefined,
+    path: [],
+  };
```

Comment:
Prettier breaks those into multiple lines even if the object is short enough for one line when the input contains a line break. Would it be possible to try to keep them in one line? Or would that require a strange case distriction in the code generator?